lib/core/test/unit/Cardano/Wallet/NetworkSpec.hs

```diff
@@ -37,5 +58,44 @@ spec = do
         testShow $ ErrPostTxBadRequest mempty
         testShow $ ErrPostTxProtocolFailure mempty
 
+    followSpec
+
+followSpec :: Spec
+followSpec =
+    describe "follow" $ do
+        it "retries on exceptions, but not too often" $ do
+            tvar <- newTVarIO []
+            let tr = traceInTVarIO tvar
+            let getHeader = header
+            let advance _blocks _h = return $ Continue @()
+            let rollback _slot = return $ Continue @()
+            void $ race (threadDelay $ 10 * second)
+                (follow mockNetworkLayer tr [] advance rollback getHeader)
+            errors <- mapMaybe (unMsg . loContent) <$> readTVarIO tvar
+            case length errors of
+                x | x == 0 -> expectationFailure "no error was logged"
+                  | x > 10 -> expectationFailure "too many errors/retries"
```

Comment:
That bound might be _too strict_, don't you think? I mean, with the backoff, we really expect at most 3 log lines since it follows a basic exponential backoff:

- 0
- delay0=1s
- 2*delay0=2s
- 4*delay0=8s

If we look at cumulative delays, the 4th one is shorter than the 10s in the test, so it shouldn't even be logged. So in practice, we really don't expect more than 3 times (or 4 to really be safe with delays) the log line (although this makes assumption on the underlying implementation). Would be better if the retrying policy was given as an argument but I think we can live with this assumption? 

What do you think?

Reply:
Going with an `== 4` check as discussed 🤷‍♂ 

This guarantees that the test fails even with constant delay ("no backoff?").